src/test/java/com/google/cloud/tools/managedcloudsdk/internal/download/StandardDownloaderTest.java

```diff
@@ -0,0 +1,178 @@
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+
+public class StandardDownloaderTest {
+
+  @Rule public TemporaryFolder tmp = new TemporaryFolder();
+
+  @Test
+  public void testValidateAndCreateFiles_targetDirectoryIsFile()
+      throws IOException, InterruptedException {
+    Path fileWithBadParent = tmp.newFile().toPath().resolve("xyz");
+
+    StandardDownloader downloader = new StandardDownloader(null, fileWithBadParent, null, null);
+    try {
+      downloader.call();
+      Assert.fail("NotDirectoryException expected but not thrown");
+    } catch (NotDirectoryException nde) {
+      // pass
+    }
+  }
+
+  @Test
+  public void testValidateAndCreateFiles_fileAlreadyExists()
+      throws IOException, InterruptedException {
+    Path fileThatExists = tmp.newFile().toPath();
+
+    StandardDownloader downloader = new StandardDownloader(null, fileThatExists, null, null);
+    try {
+      downloader.call();
+      Assert.fail("FileAlreadyExists expected but not thrown");
+    } catch (FileAlreadyExistsException faee) {
+      // pass
+    }
+  }
+
+  @Test
+  public void testDownloadURL_createsNewDirectory() throws IOException, InterruptedException {
+    Path destination = tmp.getRoot().toPath().resolve("dir-to-create").resolve("destination-file");
+    Path testSourceFile = createTestRemoteResource(1);
+    URL fakeRemoteResource = testSourceFile.toUri().toURL();
+
+    StandardDownloader downloader =
+        new StandardDownloader(fakeRemoteResource, destination, null, null);
+
+    Path downloaderDestination = downloader.call();
+    Assert.assertEquals(destination, downloaderDestination);
+  }
+
+  @Test
+  public void testDownloadURL_worksWithNullProgressListener()
+      throws IOException, InterruptedException {
+
+    Path destination = tmp.getRoot().toPath().resolve("destination-file");
+    Path testSourceFile = createTestRemoteResource(1);
+    URL fakeRemoteResource = testSourceFile.toUri().toURL();
+
+    StandardDownloader downloader =
+        new StandardDownloader(fakeRemoteResource, destination, null, null);
+
+    Path downloaderDestination = downloader.call();
+    Assert.assertEquals(destination, downloaderDestination);
+  }
+
+  @Test
+  public void testDownloadURL() throws IOException, InterruptedException {
+    Path destination = tmp.getRoot().toPath().resolve("destination-file");
+    Path testSourceFile = createTestRemoteResource(StandardDownloader.BUFFER_SIZE * 10 + 1);
+    URL fakeRemoteResource = testSourceFile.toUri().toURL();
+    DownloadProgressListener mockListener = Mockito.mock(DownloadProgressListener.class);
+
+    StandardDownloader downloader =
+        new StandardDownloader(fakeRemoteResource, destination, null, mockListener);
+
+    Path downloaderDestination = downloader.call();
+    Assert.assertEquals(destination, downloaderDestination);
+
+    Assert.assertTrue(FileUtils.contentEquals(destination.toFile(), testSourceFile.toFile()));
```

Comment:
could simply Files.readAllBytes and compare the arrays to avoid the dependency.

Reply:
Makes sense.